paddle/fluid/framework/new_executor/instruction/cond_instruction.cc

```diff
@@ -48,45 +48,31 @@ std::vector<pir::Value> GetYiedOpInputs(pir::Block* block) {
       }
     }
   }
-
   return vec_res;
 }
 
-void GetInputIds(
-    pir::Operation* op,
-    Scope* inner_scope,
-    const std::unordered_map<::pir::Value, std::string>& value_2_var_name,
-    const std::map<std::string, int>& var_name_2_id,
-    const std::unordered_map<const paddle::framework::Variable*, std::string>&
-        variable_2_var_name,
-    std::unordered_map<pir::Value, std::vector<int>>* input_ids) {
+void GetInputIds(pir::Operation* op,
+                 const ValueExecutionInfo& value_exec_info,
+                 std::unordered_map<pir::Value, std::vector<int>>* input_ids) {
   for (size_t i = 0; i < op->num_operands(); i++) {
     pir::Value value = op->operand_source(i);
-    if (value) {
-      PADDLE_ENFORCE_NE(
-          value_2_var_name.find(value),
-          value_2_var_name.end(),
+    if (value && value.type()) {
+      PADDLE_ENFORCE_EQ(
+          value_exec_info.HasValue(value),
+          true,
           phi::errors::PreconditionNotMet(
               "input should in name map, [%d] 'th input of [%s] op",
               i,
               "if op"));
-      std::vector<int> inputs_id = GetValueIds(value,
-                                               inner_scope,
-                                               value_2_var_name,
-                                               var_name_2_id,
-                                               variable_2_var_name);
+      std::vector<int> inputs_id = GetValueIds(value, value_exec_info);
```

Comment:
```suggestion
      input_ids->emplace(value, GetValueIds(value, value_exec_info));
```
这里有input_ids和 inputs_id，感觉可以只保留最终的，不需要中间tmp变量，也就不需要解决命名问题了

Reply:
done